code/src/main/java/habit_mode/view/codebehind/SudokuScreenCodeBehind.java

```diff
@@ -0,0 +1,164 @@
+
+        app_stage.show();
+        
+    }
+
+    @FXML
+    void numberButtonClicked(ActionEvent event) {
+
+    }
+
+    @FXML
+    void settingsButtonClicked(ActionEvent event) {
+
+    }
+
+    @FXML
+    void sudokuButtonSelected(ActionEvent event) {
+
+    }
+
+    @FXML
+    void initialize() {
+        this.sudokuBoard = new TextField[9][9];
+        assert this.noSelectedHabitLabel != null : "fx:id=\"noSelectedHabitLabel\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.habitListButton != null : "fx:id=\"habitListButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.settingsButton != null : "fx:id=\"settingsButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.coinsLabel != null : "fx:id=\"coinsLabel\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.backButton != null : "fx:id=\"backButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sudokuScreenLabel != null : "fx:id=\"sudokuScreenLabel\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sudokuButton != null : "fx:id=\"sudokuButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.oneButton != null : "fx:id=\"oneButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.twoButton != null : "fx:id=\"twoButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.threeButton != null : "fx:id=\"threeButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.fourButton != null : "fx:id=\"fourButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.fiveButton != null : "fx:id=\"fiveButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sixButton != null : "fx:id=\"sixButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.sevenButton != null : "fx:id=\"sevenButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.eightButton != null : "fx:id=\"eightButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        assert this.nineButton != null : "fx:id=\"nineButton\" was not injected: check your FXML file 'SudokuScreen.fxml'.";
+        this.addTextFields();
+
+    }
+
+    void addTextFields() {
+        for (int row = 0; row < 9; row++) {
+            for (int column = 0; column < 9; column++) {
+                TextField field = new TextField();
+                field.setPrefHeight(50);
+                field.setPrefWidth(50);
+                field.setAlignment(Pos.CENTER);
+                field.setFont(Font.font(20));
+                this.sudokuBoard[row][column] = field;
+                this.sudokuPane.add(field, row, column);
+            }
+        }
+    }
```

Comment:
There needs to be some padding between the cells, otherwise the cells will overlap slightly. Don't know if the GridPane supports this or not.